value/value.go

```diff
@@ -191,9 +195,14 @@ func NewDoubleValue(raw string) (*DoubleValue, error) {
 		return nil, errors.InvalidArgument(fmt.Errorf("unsupported value type: %w ", err).Error())
 	}
 
+	f, err := strconv.ParseFloat(raw, 64)
+	if err != nil {
+		return nil, errors.InvalidArgument(fmt.Errorf("unsupported value type: %w ", err).Error())
+	}
+
 	i := &DoubleValue{
 		Double:   val,
-		asString: raw,
+		asString: strconv.FormatFloat(f, 'f', -1, 64),
```

Comment:
Do we have test for this?

Reply:
Sorry missed this, we do have integration tests for it.